### PR TITLE
Migration to Gradle 7.4.2 & AGP 7.2.1

### LINKED ISSUE #474: Update Gradle
Gradle is outdated on most of the sample apps and they cannot be built on JDK 11 or 17.

### PR DESCRIPTION
**Pull request by syslogic**
Updated projects to build with Android Studio Chipmunk Patch 1.
Also migrated `buildscript` to `plugins` block; Jetifier disabled.
The `HdrViewfinder` RS example will be obsolete with Gradle 8.0

Commit [b269dc1](https://github.com/android/camera-samples/pull/489/commits/b269dc18c3d313d1de7f0e2734e71059611a8d3a) closes #474.

### CODE REVIEW

**Comment by donovanfm** [comment]
Thank you for the pull request, syslogic! I'll review these changes and get back to you soon.

File: Camera2Basic/app/build.gradle [hunk 1]
```diff
@@ -14,29 +16,32 @@
...
+        minSdk 21
+        targetSdk 32
         versionCode 1
         versionName "1.0.0"
     }
 
     compileOptions {
-        sourceCompatibility rootProject.ext.java_version
-        targetCompatibility rootProject.ext.java_version
+        sourceCompatibility JavaVersion.VERSION_11
```
Comment (donovanfm) [inline Camera2Basic/app/build.gradle:38]: Why hard code the Java version here? If you're trying to update from v8 to v11, can you just update it in one place in the root level build.gradle file? (Here and throughout the other examples)

**Review by donovanfm — CHANGES_REQUESTED** [review]
I've tested all the samples, and they're working great (and the Gradle alerts to upgrade are removed – good stuff!). I just have one comment about the Java version changes you made.

File: Camera2Basic/app/build.gradle (hunk 1 above)
Comment (Ismul32) [reply Camera2Basic/app/build.gradle:38]: []()**____**

Comment (syslogic) [reply Camera2Basic/app/build.gradle:38]: All the other modules have that hard-coded, too... and the reason is, that there is no more `buildscript.ext`. Would need to add the `buildscript` block back in, only to define `ext`. Just combed through once again and updated some more; all the latest/stable, while possible.

**Comment by syslogic** [comment]
I just have found out by merge conflicts, that they've ripped off my whole pull request - and not even changed the requested changes, while doing so... therefore, the only sane decisions is to halt all contributions - please assign your so called "experts", my time has value.

**Comment by syslogic** [comment]
It appears to me as my authorship had been infringed upon, which I find highly unprofessional.

> Whenever a non-Googler wants to submit a patch to a Google open source project, they must first sign a Contributor License Agreement (CLA). **This allows the contributor to retain their ownership in the code submitted** while granting Google the necessary legal rights to use that contribution

**Pull request merged**